src/values/instruction_value.rs

```diff
@@ -38,6 +38,8 @@ pub enum InstructionOpcode {
     CleanupRet,
     ExtractElement,
     ExtractValue,
+    #[cfg(any(feature = "llvm8-0"))]
+    FNeg,
```

Comment:
Does `llvm_versions` not yet work on enum variants? :( Wish proc macros would keep advancing
Although it's verbose, I would prefer this to be `#[cfg(not(any(feature = "llvm3-6", ...., "llvm7-0")))]` so that it's future proofed. Same for the ones later in this file.

Reply:
I think there was a reason it didn't work, but I can't recall if I actually tried to make it work or not. I'll see if that's doable, otherwise I'll make the change you've described

Reply:
Any luck with this?

Reply:
@TheDan64 Sorry about the delay, had my hands full the past couple days, I'll try and get to this tonight or in the morning!

Reply:
No problem!

Reply:
@TheDan64 I've pushed a commit which addresses this by extending the places `llvm_versions` can be used, and adds a new macro `llvm_enum` which I think may come in handy, in certainly cleaned up the implementation of `InstructionOpcode` nicely. Mostly that part was motivated by the problem of not being able to decorate expressions with attributes (on stable anyway), as well as not being able to do the same for match arms, even when they do nothing but generate `#[cfg(..)]` attributes in their place.

When that stuff makes it into stable, then the need for `#[llvm_versioned_item]` goes away (which isn't used yet, but I suspect it either could be, or will be eventually). The `llvm_enum` attribute still has the benefit of automating the boilerplate for conversions between LLVM enums and Inkwell enums like in the case of `InstructionOpcode`, but not sure how common that really is in practice, so perhaps it could be removed eventually as well.

Anyway, hope that makes maintaining a few of these things a bit less annoying :)

Reply:
`llvm_enum` looks really neat! This is pretty awesome!

> not sure how common that really is in practice, so perhaps it could be removed eventually as well.

This is actually really common, we already have a `macro_rules` macro for this already (though I think your version is better) so maybe we should replace that with this?

IE see https://github.com/TheDan64/inkwell/blob/e1854817e5b8814ad7776a1839c57392596f74e0/src/comdat.rs#L8-L22 but there are a bunch more.

One important aspect that should work in yours (it might already since you track attrs) is applying doc comments like `enum_rename` does

Reply:
I I think we can extend the macro to cover the usages of `enum_rename`, may be better to do that in its own PR though. 

Otherwise, the macro preserves all attributes IIRC, but I will confirm that when I'm at my laptop here in a bit.

Reply:
Confirmed, doc comments on the enum variants are preserved (as are any other attributes I expect). Let me know if there is anything else outstanding!